src/Swarm/TUI/Model/UI.hs

```diff
@@ -270,6 +281,7 @@ initUIState showMainMenu cheatMode = liftIO $ do
       , _uiError = Nothing
       , _uiModal = Nothing
       , _uiGoal = Nothing
+      , _uiAchievements = M.fromList $ map (\x -> (x ^. achievement, x)) achievements
```

Comment:
```suggestion
      , _uiAchievements = M.fromList $ map (view achievement &&& id) achievements
```

I find this clearer, but YMMV.  Feel free to use or ignore this suggestion as you wish.